pandas/core/dtypes/missing.py

```diff
@@ -321,6 +324,36 @@ def _isna_string_dtype(values: np.ndarray, inf_as_na: bool) -> npt.NDArray[np.bo
     return result
 
 
+def _isna_recarray_dtype(values: np.recarray, inf_as_na: bool) -> npt.NDArray[np.bool_]:
+    result = np.zeros(values.shape, dtype=bool)
+    for i, record in enumerate(values):
+        does_record_contain_nan = np.zeros(len(record), dtype=bool)
+        if inf_as_na:
```

Comment:
Instead of `if inf_as_na` everywhere could you create a `checker` function initially like

```
if inf_as_na:
    is_missing = lambda x: np.isnan(x) or np.isinf(x)
else:
    is_missing = lambda x: np.isnan(x)
```

And use that in the loop?

Reply:
@mroeschke I rewrote the function to increase readbility. What do you think about the change?